src/main/java/org/prebid/server/bidder/ix/IxBidder.java

```diff
@@ -194,21 +200,46 @@ private static Publisher modifyPublisher(Publisher publisher, String siteId) {
         }
     }
 
-    private static List<BidderBid> extractBids(BidResponse bidResponse, BidRequest bidRequest) {
+    private List<BidderBid> extractBids(BidResponse bidResponse, BidRequest bidRequest) {
         return bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())
                 ? Collections.emptyList()
                 : bidsFromResponse(bidResponse, bidRequest);
     }
 
-    private static List<BidderBid> bidsFromResponse(BidResponse bidResponse, BidRequest bidRequest) {
+    private List<BidderBid> bidsFromResponse(BidResponse bidResponse, BidRequest bidRequest) {
         return bidResponse.getSeatbid().stream()
                 .map(SeatBid::getBid)
                 .flatMap(Collection::stream)
-                .map(bid -> prepareBid(bid, bidRequest))
-                .map(bid -> BidderBid.of(bid, getBidType(bid.getImpid(), bidRequest.getImp()), bidResponse.getCur()))
+                .map(bid -> toBidderBid(bid, bidRequest, bidResponse))
                 .collect(Collectors.toList());
     }
 
+    private BidderBid toBidderBid(Bid bid, BidRequest bidRequest, BidResponse bidResponse) {
+        final BidType bidType = getBidType(bid.getImpid(), bidRequest.getImp());
+
+        Bid.BidBuilder bidBuilder = bid.toBuilder();
```

Comment:
Can be final

Reply:
Creating builder is not a cheap operation. Can we update logic to create builder only in case if we really need it?

Reply:
Still can be final